### ISSUE TITLE
All methods are deprecated in high Sierra

### ISSUE THREAD

**Issue by windgazer** [gaomd/lock-screen-app#2]
Just a quick note, as I realised you've based your code on my solution:
None of the methods are applicable to OSX High Sierra (10.13).

However, there's an up-side. Apple finally woke up, locking screen is now available as ctrl-meta-Q (or under the 'apple' menu...)

**Comment by gaomd** [comment]
Yes, thanks for your original solution, I would deprecate this repo and update the README to guide users to use the built-in Control-Command-Q method.

**Comment by rdp** [comment]
A few here: https://stackoverflow.com/q/1976520/32453 seem to work...

**Comment by windgazer** [comment]
@rdp Although there's 1 possible candidate in there that I have not tried before, the question is, why would we? For 10.12 and older, this repository contains working methods. For 10.13, OSX provides a solution out of the box.
Am I overlooking a use-case where the built-in lock-screen does not suffice and a custom application would be needed?

**Comment by rdp** [comment]
True.  Only thing I can think of is wanting to use spotlight for it instead
of memorizing new keystrokes, but I suppose new keystrokes aren't too bad :)

**Comment by gaomd** [comment]
I've made an update for macOS 10.13 HS and deprecated earlier versions, results in easier documents and simplified building process. 

Also recommended the shortcut method in the first place, so nobody would use repo anymore in theory.

However, I think this repo should still functional for providing a way to anyone insists on locking screen from the Spotlight.

**Comment by Muppetteer** [comment]
I have a couple of good use cases in High Sierra that don't involve Spotlight.

This app is golden for anyone who wants to automatically lock the screen immediately after logging in. I have a headless mac mini server running a couple of things that won't run as a daemon and require logging in to run - eg Plex Media Server and possibly Apple's content caching server (it at least needs the mac to be awake, not sure about logged in).

So I set the machine to auto login a user on boot and would prefer to then have the screen locked so anyone who tries to access it needs to know the password. For some reason that I can't be bothered trying to diagnose anymore my screen saver won't kick in so I can't rely on that to lock the screen for me. So I put this app in the login items for the user that is automatically logged in and my problem is solved: the user auto logs in and starts required applications, meanwhile the screen is locked. It is not infallible for a determined attacker but its good enough for my use at home.

Also as the machine is headless I control it via screen sharing. While screen sharing, the lock screen keyboard shortcut doesn't work as it locks my host machine instead of the server. So I added a shortcut to the app on the server's dock and I have a one click lock screen button for when I end my screen sharing session, saving me a whole click each time vs Apple's two click menu option. Hooray.

So thanks.

**Comment by windgazer** [comment]
Just out of curiosity, is that mac mini running on High Sierra? Ifso, which method are you using, because when I tested this, none of the existing methods I implemented locked the screen... Which was good, cause otherwise I'd never have found out about the new shortcut.

This is just to slate my curiosity as my macbook pro died out of warranty and the current line-up I just can't justify for myself.
